lib/group-update.js

```diff
@@ -0,0 +1,331 @@
+const ERR = require('async-stacktrace');
+const util = require('util');
+const _ = require('lodash');
+const streamifier = require('streamifier');
+const csvtojson = require('csvtojson');
+const namedLocks = require('../lib/named-locks');
+
+const logger = require('./logger');
+const serverJobs = require('./server-jobs');
+const sqldb = require('@prairielearn/prairielib/sql-db');
+const sqlLoader = require('@prairielearn/prairielib/sql-loader');
+
+const sql = sqlLoader.loadSqlEquiv(__filename);
+
+module.exports = {
+
+    /**
+     * Update question instance scores from a CSV file.
+     *
+     * @param {number} assessment_id - The assessment to update.
+     * @param {csvFile} csvFile - An object with keys {originalname, size, buffer}.
+     * @param {number} user_id - The current user performing the update.
+     * @param {number} authn_user_id - The current authenticated user.
+     * @param {function} callback - A callback(err, job_sequence_id) function.
+     */
+    uploadInstanceGroups(assessment_id, csvFile, user_id, authn_user_id, callback) {
```

Comment:
When I upload an invalid CSV (say, an image file), I get an error. I believe this is coming from this code.

```
TypeError: callback is not a function
    at /Users/nathan/git/PrairieLearn/lib/server-jobs.js:180:34
    at Query.callback (/Users/nathan/git/PrairieLearn/node_modules/@prairielearn/prairielib/lib/sql-db.js:429:17)
    at Query.handleReadyForQuery (/Users/nathan/git/PrairieLearn/node_modules/pg/lib/query.js:129:12)
    at Connection.<anonymous> (/Users/nathan/git/PrairieLearn/node_modules/pg/lib/client.js:266:19)
    at Connection.emit (events.js:198:13)
    at Connection.EventEmitter.emit (domain.js:448:20)
    at parse (/Users/nathan/git/PrairieLearn/node_modules/pg/lib/connection.js:109:10)
    at Parser.parse (/Users/nathan/git/PrairieLearn/node_modules/pg-protocol/dist/parser.js:42:17)
    at Socket.stream.on (/Users/nathan/git/PrairieLearn/node_modules/pg-protocol/dist/index.js:8:42)
    at Socket.emit (events.js:198:13)
    at Socket.EventEmitter.emit (domain.js:448:20)
    at addChunk (_stream_readable.js:287:12)
    at readableAddChunk (_stream_readable.js:268:11)
    at Socket.Readable.push (_stream_readable.js:223:10)
    at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
```

Reply:
There were some bugs in our previous version (group-update.js calls SQL many times), but now it was replaced with calling a stored procedure only 1 time. I tested upload CSV many times, and it seems working fine.

Reply:
I'll add a filter to prevent invalid file and also return a "more friendly" message to the instructor